dask_ml/model_selection/_incremental.py

```diff
@@ -757,7 +859,58 @@ def _get_params(self):
         else:
             return ParameterSampler(self.parameters, self.n_initial_parameters)
 
+    def _stop_on_plateau(self, info, instructions):
+        _best_plateau = {}
+        out = {}
+        for k, steps in instructions.items():
+            records = info[k]
+            current_calls = records[-1]["partial_fit_calls"]
+            if self.max_iter and current_calls >= self.max_iter:
+                out[k] = 0
+            elif self.patience and current_calls >= self.patience:
+                plateau = [
+                    h["score"]
+                    for h in records
+                    if current_calls - h["partial_fit_calls"] <= self.patience
```

Comment:
It is if every `partial_fit` call is coupled with a `score` call, but it may not be in general. It is for TimeDecaySearchCV, but it may not be for different implementations of `_adapt`.

Reply:
> we could send the instructions "model 1 needs 2 additional calls", so we would have a kind of "gap" in our history where we only see the score for partial fits n and n + 2? Is that correct?

Correct.

> in theory

#221 does this: `_additional_calls` tends to specify more than 1 `partial_fit` call.

I suppose a cleaner implementation would be

``` python
plateau = [h["score"] for h in self.history[-self.patience:] if "score" in h]
```

Reply:
So AFAICT, in https://github.com/dask/dask-ml/blob/90fbe69771926b7e43a5bef99d81e268dad63e7e/dask_ml/model_selection/_incremental.py#L231-L233 is the only place where we do something once *per partial_fit_call*. Everything else looks like it's done once per `_fit` call. So right now I don't see how we would end up with an entry in history that doesn't have a score. It seems like history will necessary only report models and scores that are the last in a "batch".

Reply:
Yup, you're right. Only the final `meta` is appended to `info[ident]`, which definitely has `score` in it at https://github.com/dask/dask-ml/blob/90fbe69771926b7e43a5bef99d81e268dad63e7e/dask_ml/model_selection/_incremental.py#L208

That took me by surprise.

Reply:
I think this conversation is resolved. We've clarified that using `history[-self.patience:]` will define patience to be number of score calls. This PR has implemented checking to see if the number of `partial_fit` calls is in the most recent `patience` calls.

Reply:
@mrocklin does this sound OK to you? I think users should be thinking in terms of `partial_fit_calls`, not in terms of calls to `score` (which only happens once per "batch").

@stsievert can you think of a short way to explain this that would be suitable for the docstring describing `patience`?

Reply:
> I think users should be thinking in terms of partial_fit_calls, not in terms of calls to score

Fine with me